### PR TITLE
Limit report filter counters

### PR DESCRIPTION
**Pull request by csordasmarton**
Limit results of the report filter counter functions.

This commit might solves #1053 issue.

### CODE REVIEW

File: api/v6/report_server.thrift [hunk 1]
```diff
@@ -10,6 +10,7 @@ namespace py codeCheckerDBAccess_v6
 namespace js codeCheckerDBAccess_v6
 
 const i64 MAX_QUERY_SIZE = 500
+const i64 DEFAULT_QUERY_FILTER_SIZE = 10
```
Comment (bruntib) [inline api/v6/report_server.thrift]: Is this constant necessary?

File: api/v6/report_server.thrift [hunk 2]
```diff
@@ -262,7 +264,8 @@ service codeCheckerDBAccess {
   // counts will be calculated for all of the available runs.
   // PERMISSION: PRODUCT_ACCESS
   RunReportCounts getRunReportCounts(1: list<i64>    runIds,
-                                     2: ReportFilter reportFilter)
+                                     2: ReportFilter reportFilter,
+                                     3: i64          limit)
```
Comment (bruntib) [inline api/v6/report_server.thrift]: These limit variables could be optional.

Comment (gyorb) [reply api/v6/report_server.thrift]: Do we want to introduce offset too?

File: libcodechecker/server/api/report_server.py [hunk 3]
```diff
@@ -1434,12 +1443,15 @@ def getCheckerMsgCounts(self, run_ids, report_filter, cmp_data):
...
                                                  func.count(q.c.bug_id)) \
-                    .group_by(q.c.checker_message).all()
+                    .group_by(q.c.checker_message)
             else:
                 checker_messages = q.group_by(Report.checker_message,
-                                              Report.severity).all()
+                                              Report.severity)
+
+            if limit:
+                checker_messages = checker_messages.limit(limit).all()
```
Comment (gyorb) [inline libcodechecker/server/api/report_server.py]: is the `.all()` needed here?

File: api/v6/report_server.thrift (hunk 2 above)
Comment (csordasmarton) [reply api/v6/report_server.thrift]: It seems to me that thrift doesn't support optional function arguments. The compiler gives me the following warning: `optional keyword is ignored in argument lists`. (https://svn.apache.org/viewvc/incubator/thrift/trunk/compiler/cpp/src/thrifty.yy?view=markup&pathrev=665255)

**Comment by bruntib** [comment]
When clicking on the message filter options, the tooltip overflows the window at the bottom and those parts can be reached with an extra scrollbar.
This happens because of the lazy loading of its content. Maybe this can be solved by invoking an extra renderer function on the tooltip.

**Pull request merged**